govcd/edgegateway.go

```diff
@@ -1144,20 +1144,99 @@ func (egw *EdgeGateway) UpdateLBGeneralParams(enabled, accelerationEnabled, logg
 		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
 	}
 	_, err = egw.client.ExecuteRequestWithCustomError(httpPath, http.MethodPut, types.AnyXMLMime,
-		"error while updating load balancer application rule : %s", currentLb, &types.NSXError{})
+		"error while updating load balancer config: %s", currentLb, &types.NSXError{})
 	if err != nil {
 		return nil, err
 	}
 
 	// Retrieve configuration after update
 	updatedLb, err := egw.GetLBGeneralParams()
 	if err != nil {
-		return nil, fmt.Errorf("unable to retrieve load balancer after update: %s", err)
+		return nil, fmt.Errorf("unable to retrieve load balancer config after update: %s", err)
 	}
 
 	return updatedLb, nil
 }
 
+// GetFwGeneralParams retrieves firewall configuration and can be used
+// to alter master configuration options. These are 3 fields only:
+// FwGeneralParamsWithXml.Enabled, FwGeneralParamsWithXml.DefaultPolicy.LoggingEnabled and
+// FwGeneralParamsWithXml.DefaultPolicy.Action
+func (egw *EdgeGateway) GetFwGeneralParams() (*types.FwGeneralParamsWithXml, error) {
+	if !egw.HasAdvancedNetworking() {
+		return nil, fmt.Errorf("only advanced edge gateway support firewall configuration")
+	}
+
+	httpPath, err := egw.buildProxiedEdgeEndpointURL(types.EdgeFirewallPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
+	}
+
+	firewallConfig := &types.FwGeneralParamsWithXml{}
+	_, err = egw.client.ExecuteRequest(httpPath, http.MethodGet, types.AnyXMLMime,
+		"unable to read firewall configuration: %s", nil, firewallConfig)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return firewallConfig, nil
+}
+
+// UpdateFwGeneralParams allows to update firewall configuration.
+// It accepts three fields (Enabled, DefaultLoggingEnabled, DefaultAction) and uses
+// them to construct types.FwGeneralParamsWithXml without altering other options to prevent config
+// corruption.
+// They are represented in firewall configuration page in the UI.
+func (egw *EdgeGateway) UpdateFwGeneralParams(enabled, defaultLoggingEnabled bool, defaultAction string) (*types.FwGeneralParamsWithXml, error) {
```

Comment:
Change func name to config from params if agree on that.

Reply:
I prefer `Upfdate...` as it makes clear that it is modifying the entity.

Reply:
`UpdateFwGeneralConfig` it is